site/src/pages/ManagementSettingsPage/GroupsPage/GroupsPageView.tsx

```diff
@@ -0,0 +1,208 @@
+                        hover
+                        key={group.id}
+                        data-testid={`group-${group.id}`}
+                        tabIndex={0}
+                        onClick={() => {
+                          navigate(group.name);
+                        }}
+                        onKeyDown={(event) => {
+                          if (event.key === "Enter") {
+                            navigate(group.name);
+                          }
+                        }}
+                        css={styles.clickableTableRow}
+                      >
+                        <TableCell>
+                          <AvatarData
+                            avatar={
+                              <GroupAvatar
+                                name={group.display_name || group.name}
+                                avatarURL={group.avatar_url}
+                              />
+                            }
+                            title={group.display_name || group.name}
+                            subtitle={`${group.members.length} members`}
+                          />
+                        </TableCell>
+
+                        <TableCell>
+                          {group.members.length === 0 && "-"}
+                          <AvatarGroup
+                            max={10}
+                            total={group.members.length}
+                            css={{ justifyContent: "flex-end" }}
+                          >
+                            {group.members.map((member) => (
+                              <UserAvatar
+                                key={member.username}
+                                username={member.username}
+                                avatarURL={member.avatar_url}
+                              />
+                            ))}
+                          </AvatarGroup>
+                        </TableCell>
+
+                        <TableCell>
+                          <div css={styles.arrowCell}>
+                            <KeyboardArrowRight css={styles.arrowRight} />
+                          </div>
+                        </TableCell>
+                      </TableRow>
+                    ))}
+                  </Cond>
+                </ChooseOne>
+              </TableBody>
+            </Table>
+          </TableContainer>
+        </Cond>
+      </ChooseOne>
+    </>
+  );
+};
+
+const TableLoader = () => {
+  return (
+    <TableLoaderSkeleton>
+      <TableRowSkeleton>
+        <TableCell>
+          <div css={{ display: "flex", alignItems: "center", gap: 8 }}>
+            <AvatarDataSkeleton />
+          </div>
+        </TableCell>
+        <TableCell>
+          <Skeleton variant="text" width="25%" />
+        </TableCell>
+        <TableCell>
+          <Skeleton variant="text" width="25%" />
+        </TableCell>
+      </TableRowSkeleton>
+    </TableLoaderSkeleton>
+  );
+};
+
+const styles = {
+  clickableTableRow: (theme) => ({
```

Comment:
I think we have a hook called "useClickableRow" that applies all the styles and events for clickable table rows.